src/views/Viewer.vue

```diff
@@ -707,6 +714,18 @@ export default {
 		onClose() {
 			this.Viewer.onClose()
 		},
+
+		async onDelete() {
+			const url = this.root + this.currentFile.filename
+			await axios.delete(url)
+			if (this.hasPrevious) {
+				this.previous()
+				const currentIndex = this.fileList.findIndex(file => file.basename === this.currentFile.basename)
+				this.fileList = this.fileList.splice(0, currentIndex)
```

Comment:
I'm pretty sure this line will also remove all further elements standing after the one we want to remove.
This should be enough:
`this.fileList.splice(currentIndex, 1);`

Test: https://jsfiddle.net/wt4yvkuz/

Reply:
Thanks!